ruby/lib/minitest/queue.rb

```diff
@@ -103,6 +147,12 @@ def run_from_queue(reporter, *)
       queue.poll do |example|
         result = example.run
         failed = !(result.passed? || result.skipped?)
+
+        if failed && example.flaky?
+          result.mark_as_flaked!
```

Comment:
We only mark as flaked in case of failure, we could do it all the time, I have no preference TBH. Thoughts?

Reply:
The external systems that determine whether something is flaky or not do have to keep track of this stuff, so there's no technical reason for us report it as flaky.

That said, passing flaky tests are still flaky so from a conceptual point of view, I guess we should still report it as flaky.

Reply:
Indeed. I also just had some though: the junit integration in `junit` will have to be made aware as well, otherwise we'll display those failures as annotations.

Reply:
> the junit integration in junit will have to be made aware as well, otherwise we'll display those failures as annotations

I'm not sure what you mean here. Do you mean the messages in `lib/minitest/queue/junit_reporter.rb` or something else? If you do mean the junit reporter, we still need the failure messages for flaky tests.

Reply:
Oh, I see what you mean. I added some appropriate tests to demonstrate this. I suppose we'll need to create `Flaked` objects for the passes too.